src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Filters/LiquidViewFilters.cs

```diff
@@ -49,47 +55,47 @@ public static ValueTask<FluidValue> HtmlClass(FluidValue input, FilterArguments
 
         public static async ValueTask<FluidValue> NewShape(FluidValue input, FilterArguments arguments, TemplateContext context)
         {
-            if (!context.AmbientValues.TryGetValue("ShapeFactory", out dynamic shapeFactory))
+            if (!context.AmbientValues.TryGetValue("ShapeFactory", out var item) || !(item is IShapeFactory shapeFactory))
             {
```

Comment:
For infos, it can't be anything else than `IShapeFactory`, it is registered in one place like this.

            var shapeFactory = services.GetRequiredService<IShapeFactory>();
            context.AmbientValues.Add("ShapeFactory", shapeFactory);

So if the overhead is just to use `out var dynamic shapeFactory`, if we use `out var shapeFactory` and then we do a cast `(IShapeFactory)shapeFactory`, is it still okay?

Does the cast has more costs than doing `item is IShapeFactory shapeFactory`?

Reply:
Added benchmark and results about using dynamic object. Cast and checks are both fine, interface dispatch is costly (call via base class is faster), but no where near as costly as dynamic objects. I'd only use dynamics like for COM interop of non-hot path things, interface is always clearer.